source/scripts/ContentScript/stargazerRequestsProxy.ts

```diff
@@ -0,0 +1,43 @@
+import { browser, Runtime } from 'webextension-polyfill-ts';
+import {
+  StargazerEncodedProxyRequest,
+  StargazerProxyRequest,
+  StargazerEncodedProxyResponse,
+  StargazerProxyResponse,
+} from '../common/proxy-types';
+
+const isStargazerProxyRequest = (value: any, once: string): value is StargazerEncodedProxyRequest => {
+  return 'once' in value && value.once === once;
+};
+
+class StargazerRequestsProxy {
+  #once: string;
+  #port: Runtime.Port;
+
+  constructor(once: string) {
+    this.#once = once;
```

Comment:
Can we name this ID or something like that? `once` is a little confusing.

Reply:
Hah renamed, idk what I was thinking but it was originally `nonce` instead of `once`... But I like id better...